cmake/generic.cmake

```diff
@@ -306,11 +306,14 @@ function(go_library TARGET_NAME)
 
   set(dummyfile ${CMAKE_CURRENT_BINARY_DIR}/${TARGET_NAME}_dummy.c)
 
+  file(GLOB GO_SOURCE RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" "*.go")
+  set(TOUCH_CMD ${PROJ_ROOT}/paddle/scripts/build/if_newer_then_touch.sh)
   # This custom command will always run since it depends on a not
   # existing file.
   add_custom_command(
     OUTPUT dummy_rebulid_${TARGET_NAME}
-    COMMAND cmake -E touch ${dummyfile}
+    COMMAND ${TOUCH_CMD} ${dummyfile} ${GO_SOURCE}
```

Comment:
If we change this line into 

```
COMMAND ${PROJ_ROOT}/paddle/scripts/build/touch_if_newer.sh ${dummyfile} ${GO_SOURCE}
```

we can remove L310 which defines `TOUCH_CMD`.